packages/server/src/user/user.service.ts

```diff
@@ -115,34 +143,53 @@ export class UserService {
   /**
    * Generate reset token for a user and update it in the database.
    *
-   * @param params object includes: `project_id` and `email`
-   * @param reset_code a string of reset code in plain text
+   * @param params object includes: `project_id` and `email` as string
+   * @param reset_code_plain a string of reset code in plain text
    */
-  async setResetToken(params: { project_id; email }, reset_code: string) {
-    // TODO: Replace set reset token for temperary user
-    for (let idx = 0; idx < this.users.length; idx++) {
-      const user = this.users[idx];
-      if (params.project_id === user.project_id && params.email === user.email) {
-        user.reset_code = reset_code;
-        break;
+  async setResetToken(params: { project_id; email }, reset_code_plain: string): Promise<void> {
```

Comment:
```suggestion
  async setResetToken(user: User, reset_code_plain: string): Promise<void> {
```
Might be useful to pass in the user object here, since Auth Service should look up before hand

Reply:
Does it mean the auth service also have access to the database?